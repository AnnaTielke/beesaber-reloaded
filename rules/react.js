module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
