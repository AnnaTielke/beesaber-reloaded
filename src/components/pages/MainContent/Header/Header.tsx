import styled from 'styled-components';

import Spinner from '@/components/common/Spinner/SpinnerPulse';

import GoogleProfileData from './GoogleProfileData/GoogleProfileData';

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-right: 2rem;
`;

const Header = (): JSX.Element => (
  <Container>
    <GoogleProfileData />
    <Spinner />
  </Container>
);

export default Header;
