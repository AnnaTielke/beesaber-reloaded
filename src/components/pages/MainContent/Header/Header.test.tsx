import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import Header from './Header';

import type { UserData } from '@/sharedTypes/UserData';
import type { AppStatus } from '@/store/reducer/appStatusReducer';
import type { RootState } from '@/store/store';

const mockStore = configureMockStore();

const appStatus: AppStatus = {
  isLoggedIn: true,
  isLoggingIn: false,
};

const store: RootState = {
  appStatus,
  notifications: [],
  userData: {} as UserData,
};

describe('Header', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={mockStore(store)}>
        <Header />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});

test.todo('show / hide spinner when (no) fetch in progress');
test.todo('match snapshot spinner true/false');
