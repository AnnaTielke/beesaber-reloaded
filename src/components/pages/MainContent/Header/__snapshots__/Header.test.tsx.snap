// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
.c3 .ldio-do9usc7wgsf div {
  position: absolute;
  border-width: 0.3125rem;
  border-style: solid;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: ldio-do9usc7wgsf 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: ldio-do9usc7wgsf 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

.c3 .ldio-do9usc7wgsf div:nth-child(1) {
  border-color: #ffeb00;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c3 .ldio-do9usc7wgsf div:nth-child(2) {
  border-color: #0000FF;
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c3 .loadingio-spinner-ripple-rt8n616prbh {
  width: 3.8125rem;
  height: 3.8125rem;
  display: inline-block;
  overflow: hidden;
  background: #00111f;
}

.c3 .ldio-do9usc7wgsf {
  width: 100%;
  height: 100%;
  position: relative;
  -webkit-transform: translateZ(0) scale(0.61);
  -ms-transform: translateZ(0) scale(0.61);
  transform: translateZ(0) scale(0.61);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c3 .ldio-do9usc7wgsf div {
  box-sizing: content-box;
}

.c1 {
  max-height: 100px;
  max-width: 100px;
}

.c2 {
  border-radius: 1.5625rem;
  margin-right: 1rem;
  max-width: 3.125rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="profile pic"
      class="c2"
      id="profile-pic-sm"
      referrerpolicy="no-referrer"
      src="/public/imgs/m_bee.jpg"
    />
    <span>
      test
    </span>
  </div>
  <div
    class="c3"
  >
    <div
      class="loadingio-spinner-ripple-rt8n616prbh"
    >
      <div
        class="ldio-do9usc7wgsf"
      >
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
